Sources/Transaction/Outputs/PendingTransaction.swift

```diff
@@ -35,3 +35,33 @@ public struct PendingSinglePayloadTransaction {
 
 extension PendingTransaction: Equatable, Hashable {}
 extension PendingSinglePayloadTransaction: Equatable {}
+
+extension PendingSinglePayloadTransaction {
+    func isIdempotent(_ lhs: PendingSinglePayloadTransaction) -> Bool {
```

Comment:
given the meaning of idempotent, it's not 100% clear, by name alone, if this should compare for equality or inequality...   

Can we rename these to `isIdempotentlyEqual` or something like that? 

Same for the `are` functions...

Reply:
I like the name `isIdempotentlyEqual` or `isIdempotentlyEquivalent`

Reply:
I was thinking to add "with" so it would read `isIdempotent(with: lhs)` call-site:

```swift
payload.isIdempotent(with: otherPayload)
```

I think the equal qualifier is redundant with idempotent. Not sure the static functions gain much from a rename.

Reply:
changed to `isIdempotent(with: lhs)`

Reply:
`isSomething` implies test for equality to "something", `isNotSomething` is the reverse.